distribution/registry.go

```diff
@@ -59,9 +57,9 @@ func newV2Repository(repoInfo *registry.RepositoryInfo, endpoint registry.APIEnd
 	authTransport := transport.NewTransport(base, modifiers...)
 	pingClient := &http.Client{
 		Transport: authTransport,
-		Timeout:   15 * time.Second,
+		Timeout:   5 * time.Second,
 	}
-	endpointStr := strings.TrimRight(endpoint.URL, "/") + "/v2/"
+	endpointStr := endpoint.URL + "/v2/"
```

Comment:
> You've reverted #17203?

Yes, unintentionally. Since the package name has changed, these kinds of changes need to be handled manually. This one seems to have slipped through.

I've added the missing change. Thanks for the careful review!